Nim string definition conflicts with other C/C++ instances

The Oracle Instant Client code has the following definition in oratypes.h.

```c
typedef  oratext *string;
```

This conflicts with nimbase.h:

```c
typedef struct TStringDesc* string;
```

Ideally the Nim implementation should use a unique name so that it doesn't conflict with any existing C/C++ libs that are wrapped.

```nim
import oci

let username = "...".cstring
let password = "...".cstring
let server   = "...".cstring

let ptrUsername = cast[ptr OraText](username)
let ptrPassword = cast[ptr OraText](password)
let ptrServer   = cast[ptr OraText](server)

var
  xtramemsz = 0.cuint
  nimOCIEnv = 0.cuint
  nimOCIError: ptr OCIError
  PNimOCIEnv: ptr OCIEnv
  PPNimOCIEnv = PNimOCIEnv.addr
  addrXtramemsz: ptr = xtramemsz.addr
  usrmempp: pointer
  Pusrmempp = usrmempp.addr
  nimOCI_DEFAULT = cast[ub4](OCI_DEFAULT)


# proc OCIEnvInit*(envp: ptr ptr OCIEnv; mode: ub4; xtramem_sz: cuint; usrmempp: ptr pointer): sword {.importc, header: headerociap.}
var environment = OCIEnvInit(PPNimOCIEnv, nimOCI_DEFAULT, xtramemsz, Pusrmempp)

# ERROR:
# ======
# In file included from C:/code/instantclient_18_5/sdk/include/ociap.h:802:0,
#                  from C:\Users\jaredfields\nimcache\nim_Oracle_r\nim_Oracle.c:10:
# C:/code/instantclient_18_5/sdk/include/oratypes.h:311:19: error: conflicting types for 'string'
#  typedef  oratext *string;
#                    ^~~~~~
# In file included from C:\Users\jaredfields\nimcache\nim_Oracle_r\nim_Oracle.c:9:0:
# C:\nim\lib/nimbase.h:397:29: note: previous declaration of 'string' was here
#  typedef struct TStringDesc* string;
#                              ^~~~~~
```

--- a/lib/nimbase.h
+++ b/lib/nimbase.h
@@ -394,8 +394,6 @@ static N_INLINE(NI32, float32ToInt32)(float x) {
      NIM_CHAR data[(length) + 1];          \
   } name = {{length, (NI) ((NU)length | NIM_STRLIT_FLAG)}, str}
 
-typedef struct TStringDesc* string;
-
 /* declared size of a sequence/variable length array: */
 #if defined(__GNUC__) || defined(__clang__) || defined(_MSC_VER)
 #  define SEQ_DECL_SIZE /* empty is correct! */

